### PR TITLE
Cleanup login promo code & storyboard

### PR DESCRIPTION
**Pull request by nheagy**
This removes the five "promo" pages from the login storyboard and moves them to pure code layout. It was kind of weird having five copies of a very basic view controller in the storyboard, and now they're all in one place, and in code.

To test:
- ensure that the promo screens look exactly as they did before. I did that, and here are my results:

Before: SE | After: SE
| --- | --- |
iPhone SE
| simulator screen shot jul 24 2017 4 17 14 pm | simulator screen shot jul 24 2017 4 25 27 pm |
iPhone 6
| simulator screen shot jul 24 2017 4 14 41 pm | simulator screen shot jul 24 2017 4 24 43 pm |
iPad Air Portrait
| simulator screen shot jul 24 2017 4 25 57 pm | simulator screen shot jul 24 2017 4 15 46 pm |
iPad Air Landscape
| simulator screen shot jul 24 2017 4 26 08 pm |simulator screen shot jul 24 2017 8 49 21 pm |

Needs review: @frosty
James, want to take a look at this one?

### CODE REVIEW

File: WordPress/Classes/ViewRelated/NUX/LoginProloguePromoViewController.swift [hunk 1]
```diff
@@ -2,88 +2,144 @@ import UIKit
...
+        animationView.leadingAnchor.constraint(equalTo: animationHolder.leadingAnchor).isActive = true
+        animationView.trailingAnchor.constraint(equalTo: animationHolder.trailingAnchor).isActive = true
+        animationView.topAnchor.constraint(equalTo: animationHolder.topAnchor).isActive = true
+        animationView.bottomAnchor.constraint(equalTo: animationHolder.bottomAnchor).isActive = true
+    }
+
+
+    // MARK: promo settings
+
+    private func headlineColor() -> UIColor {
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/NUX/LoginProloguePromoViewController.swift]: Let's make this a `var` and move it onto an extension of `PromoType`.

File: WordPress/Classes/ViewRelated/NUX/LoginProloguePromoViewController.swift [hunk 2]
```diff
@@ -2,88 +2,144 @@ import UIKit
...
+        switch type {
+        case .post, .reader, .jetpack:
+            return UIColor(hexString: "204E80")
         default:
-            return "jetpack"
+            return UIColor.white
         }
     }
 
     private func headlineText() -> String {
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/NUX/LoginProloguePromoViewController.swift]: Let's make this a `var` and move it onto an extension of `PromoType`.

**Comment by frosty** [comment]
Also this:

```
2017-07-25 11:09:05.678756+0100 WordPress[98763:62039335] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x610000099a00 UIStackView:0x7ff288cd9f90.width == 0.8*UIView:0x7ff288cf9930.width   (active)>",
    "<NSLayoutConstraint:0x610000487170 UILabel:0x7ff288cda520'Publish from the park. Bl...'.width <= 600   (active)>",
    "<NSLayoutConstraint:0x60800029ee60 'UISV-canvas-connection' UIStackView:0x7ff288cd9f90.leading == UILabel:0x7ff288cda520'Publish from the park. Bl...'.leading   (active)>",
    "<NSLayoutConstraint:0x60800029eeb0 'UISV-canvas-connection' H:[UILabel:0x7ff288cda520'Publish from the park. Bl...']-(0)-|   (active, names: '|':UIStackView:0x7ff288cd9f90 )>",
    "<NSLayoutConstraint:0x60800029eff0 'UIView-Encapsulated-Layout-Width' UIView:0x7ff288cf9930.width == 768   (active)>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x610000487170 UILabel:0x7ff288cda520'Publish from the park. Bl...'.width <= 600   (active)>
```

**Comment by nheagy** [comment]
@frosty thanks for the review, I implemented all your suggestions and it looks so clean 🌟 

I was able to fix the first constraint problem you found, which I think is in `release/8.1` as well… But the second I can't recreate. Perhaps fixing the first fixed it? I dunno…

Anyway, back to you for second review!

**Comment by nheagy** [comment]
Thanks @frosty 🎉

**Pull request merged**